packages/clerk-js/src/core/clerk.ts

```diff
@@ -1953,10 +1954,11 @@ export class Clerk implements ClerkInterface {
     }
 
     if (su.status === 'complete') {
-      return this.setActive({
+      await this.setActive({
         session: su.sessionId,
         redirectUrl: redirectUrls.getAfterSignUpUrl(),
       });
+      return this.__internal_navigateToTaskIfAvailable({ redirectUrlComplete: redirectUrls.getAfterSignUpUrl() });
```

Comment:
To be clear, this doesn't impact custom flows, and developers shouldn't have to do it. 

On their custom pages, `Clerk.handleRedirectCallback` will always navigate to the pending task route.